stdlib/LinearAlgebra/src/generic.jl

```diff
@@ -1415,3 +1415,32 @@ function normalize(v::AbstractVector, p::Real = 2)
         return T[]
     end
 end
+
+"""
+    eachrow(A::AbstractVecOrMat)
+
+Create a generator iterating over views of A's first dimension.
+See also [`eachcol`](@ref) and [`eachslice`](@ref).
+"""
+eachrow(A::AbstractVecOrMat) = (view(A, i, :) for i in axes(A, 1))
+
+
+"""
+    eachcol(A::AbstractVecOrMat)
+
+Create a generator iterating over views of A's second dimension.
+See also [`eachrow`](@ref) and [`eachslice`](@ref).
+"""
+eachcol(A::AbstractVecOrMat) = (view(A, :, i) for i in axes(A, 2))
+
+"""
+    eachslice(A; dims = 1)
+
+Create a generator iterating over views of A's nth dimension.
+See also [`eachrow`](@ref) and [`eachcol`](@ref).
+"""
+@inline function eachslice(A; dims = 1)
```

Comment:
Since this is an untyped `A`, what is the interface that it needs to follow for it to be valid to call `eachslice` on it?

Reply:
Good point: it's always easier to artificially restrict new functionality and later widen it.  Let's just assert `::AbstractArray` here.

Reply:
Done.